protocol/lib/metrics/util.go

```diff
@@ -80,3 +80,18 @@ func GetCallbackMetricFromCtx(ctx sdk.Context) string {
 
 	return DeliverTx
 }
+
+func ContextuallySetGaugeWithLabels(
+	ctx sdk.Context,
+	keys []string,
+	val float32,
+	labels []gometrics.Label,
+	allowedModes []sdk.ExecMode,
+) {
+	contextExecMode := ctx.ExecMode()
+	for _, mode := range allowedModes {
+		if contextExecMode == mode {
+			telemetry.SetGaugeWithLabels(keys, val, labels)
```

Comment:
Can you check with @jonfung-dydx if we should be using functions [here](https://github.com/dydxprotocol/v4-chain/blob/454cb0c2a9d9fa58b8f38019b6e1e4569d95c939/protocol/lib/metrics/lib.go#L10)?

Reply:
can we move these functions over to the lib.go file? and use `SetGaugeWithLabels`?

also nit: `SetGaugeWithLabelsForExecutionMode` since all our logs area already contextualized

Reply:
yeah, all those places should be migrated over into one-key 😆 
https://www.notion.so/dydx/Metrics-Guidelines-f31f02e792bc4914abac2309df02e3b0?pvs=4#7e8b21909f404372940d062dde577e0b

Reply:
ah... got it!